src/main/java/seedu/address/ui/FeedPostListPanel.java

```diff
@@ -31,8 +35,43 @@ public FeedPostListPanel(ReadOnlyFeedList feedList) {
         feedPostListView.setItems(feedPostList);
         feedPostListView.setCellFactory(listView -> new FeedPostListViewCell());
 
+        fetchPosts(feedPostList, feedList.getFeedList());
+
+        ObservableList<Feed> observableFeedList = feedList.getFeedList();
+        ListChangeListener<Feed> listener = change -> updatePosts(feedPostList, change);
+        observableFeedList.addListener(listener);
+    }
+
+    /**
+     * Detects the changes made in a `ListChangeListener.Change` object and updates the post list accordingly.
+     * If feeds are added, trigger a thread to fetch its posts. If feeds are removed, remove all its posts from the
+     * feed list.
+     * @param feedPostList Observable list of feed posts
+     * @param change Object representing a change in the feed list
+     */
+    private void updatePosts(ObservableList<FeedPost> feedPostList, ListChangeListener.Change<? extends Feed> change) {
+        change.next();
+        List<Feed> added = change.getAddedSubList().stream().filter(Objects::nonNull).collect(Collectors.toList());
+        List<Feed> removed = change.getRemoved().stream().filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (added.size() > 0) {
+            fetchPosts(feedPostList, added);
+        }
+        if (removed.size() > 0) {
+            for (Feed f : removed) {
+                feedPostList.removeIf(feedPost -> feedPost.getSource().equals(f.getName()));
+            }
+        }
+    }
+
+    /**
+     * Fires off a thread that fetches posts from a list of feeds and adds them to the post list.
+     * @param feedPostList Observable list of feed posts.
+     * @param feedList List of input feeds.
+     */
+    private void fetchPosts(ObservableList<FeedPost> feedPostList, List<Feed> feedList) {
         Runnable feedPostFetch = () -> {
-            for (Feed feed : feedList.getFeedList()) {
+            for (Feed feed : feedList) {
                 ObservableList<FeedPost> feedPosts = feed.fetchPosts();
                 Platform.runLater(() -> {
                     feedPostList.addAll(feedPosts);
```

Comment:
What if the URL provided is not accurate? What will be the response given by the application?